src/generic/datavgen.cpp

```diff
@@ -6448,12 +6457,28 @@ void wxDataViewCtrl::EnsureVisibleRowCol( int row, int column )
 
     int first = m_clientArea->GetFirstVisibleRow();
     int last = m_clientArea->GetLastFullyVisibleRow();
-    if( row < first )
+    if( row <= first )
         m_clientArea->ScrollTo( row, column );
-    else if( row > last )
-        m_clientArea->ScrollTo( row - last + first, column );
-    else
-        m_clientArea->ScrollTo( first, column );
+    else if (row > last)
+    {
+        if (!HasFlag(wxDV_VARIABLE_LINE_HEIGHT))
+            m_clientArea->ScrollTo(row - last + first, column);
+        else
+        {
+            // calculate scroll position based on last visible item
+            const int itemStart = m_clientArea->GetLineStart(row);
+            const int itemHeight = m_clientArea->GetLineHeight(row);
+            const int clientHeight = m_clientArea->GetSize().y;
+            int scrollX, scrollY;
+            GetScrollPixelsPerUnit(&scrollX, &scrollY);
+            int scrollPosY = (itemStart + itemHeight - clientHeight + scrollY - 1) / scrollY;
+            int scrollPosX = m_clientArea->GetColumnStart(column);
+            if (column != -1 && scrollX)
+                scrollPosX = scrollPosX / scrollX;
```

Comment:
Shouldn't we round up here too, i.e.
```suggestion
                scrollPosX = (scrollPosX + scrollX - 1)/ scrollX;
```
?